samples/drivers/gpio/custom_dts_binding/dts/bindings/power-output.yaml

```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2020 Martin Jäger / Libre Solar
+# SPDX-License-Identifier: Apache-2.0
+
+description: GPIO pin to switch a power output on or off
```

Comment:
I appreciate the effort that went into this sample and I think the demonstration of how to make a custom binding is worthwhile, so I am not against merging it at all.

However, it doesn't really address my concern that this is a lot of boilerplate just to define a pin. People have expressed concerns to me that moving things like "use this pin" from Kconfig to DT takes an amount of extra effort that feels unnecessary, compared to just defining a string symbol which is the GPIO device name, and an int symbol which is the pin number.

I tend to agree that this feels like a lot of extra boilerplate. I wonder if there's some way we can propose a convention for miscellaneous application-specific GPIOs for people in a hurry who don't want to go to the trouble of creating a binding just for a pin.

Thanks again for this patch.

Reply:
Yeah so there are basically 3 options:
1. With custom DTS binding as in this sample.
2. Abuse the existing `gpio-leds` binding for digital outputs (or `gpio-keys` for inputs). I guess that's what many people do (like me some time ago).
3. Manually define GPIO device name and pin number as you suggested (either via Kconfig or hardcoded).

I'm still not totally against introducing more generic DTS bindings like `gpio-outputs` and `gpio-inputs` in master branch to avoid option 2.

Reply:
A recent change to devicetree tooling allowed generation of nodes for which there was no `compatible`, but that generation did not include any properties on such nodes (which I still think is weird).

Do we have enough special-casing in the tooling to support something like:
```
/ {
  app {
    power-gpios = <&gpioa 5 GPIO_ACTIVE_HIGH>;
  };
};
```
then look it up by `DT_PATH(app, power_gpios)`?  At least for GPIO we can reasonably say that if the property ends in `-gpios` we can assume it's a GPIO phandle+specifier.  @galak?

Reply:
> Do we have enough special-casing in the tooling

That doesn't work today:

```
/* Special property macros: */
#define DT_N_S_app_REG_NUM 0
#define DT_N_S_app_IRQ_NUM 0
#define DT_N_S_app_STATUS_okay 1

/* (No generic property macros) */
```

But I think it's a pretty interesting idea, actually

Reply:
I think a sample like this is very instructive in showing how custom GPIO consumer bindings are intended to be described using DTS in Zephyr.

For the "Arduino-like" approach, I really don't see we need to add anything. I mean, there is absolutely nothing wrong with people writing non-portable, out-of-tree code using e.g. `device_get_binding("GPIO_1")` and hard-coding the pin numbers for a given GPIO. Perhaps we just need to mention that this is indeed possible within the Zephyr documentation - but at the same time mention the benefits of _not_ going down that road?

I'm afraid that even if we add tooling and/or simplified bindings for describing simple GPIOs, we will still be met with questions as to why this needs to be so complicated (even though it does not need to if you do not need the abstractions and portability).